### DISCUSSION TITLE
How to protect Livewire class when included inside blade view

### DISCUSSION THREAD

**Discussion by adrianaditech** [livewire/livewire#8190] [1. Help] [answered]
I have Livewire classes in App\Livewire which should only be accessible to logged in users.
These are not full-page Livewire components; I am including them in my regular blade views with `@livewire('component-name')` which is working fine.

My blade views are accessed via routes that are already protected with auth middleware, so only logged in users can access the blade views which are including the @livewire components.

_My question is whether this is already preventing any access to the Livewire classes for non-authenticated users?_ Again they're included only in blade views, and these views are already protected with auth middleware. Is there any possibility for anyone to access the Livewire components without being logged in.

I am asking because there are no route entries for the Livewire classes in my web.php routes that I can protect manually (Livewire handles the routing automatically for these Livewire components), so I need to be sure it's only accessible for logged in users.

**Comment by mtt786** [comment]
When a Livewire component is loaded on a page containing route-level [Authorization Middleware](https://laravel.com/docs/authorization#via-middleware), like so:
```
Route::get('/post/{post}', App\Livewire\UpdatePost::class)
    ->middleware('can:update,post');
```
Livewire will ensure those middlewares are re-applied to subsequent Livewire network requests. This is referred to as "Persistent Middleware" in Livewire's core.

Persistent middleware protects you from scenarios where the authorization rules or user permissions have changed after the initial page-load.    
[Read in depth (Livewire v3 Docs)](https://livewire.laravel.com/docs/security#middleware)

**Reply by adrianaditech** [reply]
I am not using full page livewire components so there are no routes for the livewire components.
Just including them with @livewire inside blade templates. the blade templates are behind routes that are protected with auth middleware.
Want to know if the livewire components are already protected in this situation

**Comment by sanderdewijs** [comment, marked as answer]
If there are no routes pointing to the Livewire components, they are not directly accessible.
